database/models.go

```diff
@@ -166,22 +170,23 @@ func NewNamespace(name string, versionFormat string) *Namespace {
 // dpkg is the version format of the installer package manager, which in this
 // case could be dpkg or apk.
 type Feature struct {
-	Name          string      `json:"name"`
-	Version       string      `json:"version"`
-	VersionFormat string      `json:"versionFormat"`
-	Type          FeatureType `json:"type"`
+	Name               string      `json:"name"`
+	Version            string      `json:"version"`
+	VersionFormat      string      `json:"versionFormat"`
+	Type               FeatureType `json:"type"`
+	PotentialNamespace Namespace   `json:"potentialNamespace"`
```

Comment:
Please move PotentialNamespace to LayerFeature struct, because Feature is shared by multiple layers.

Reply:
Feature listers return `database.Feature` and not `database.LayerFeature`. Can I change the return type of listers or how can I extract namespace?

Reply:
yes, please change the feature lister to return LayerFeature.